src/tracer.js

```diff
@@ -18,14 +19,29 @@ export default class Tracer {
     /**
      * Starts and returns a new Span representing a logical unit of work.
      *
-     * @param  {string|object} nameOrFields - if the given argument is a
+     * For example:
+     *
+     *     // Start a new (parentless) root Span:
+     *     var parent = Tracer.startSpan('DoWork');
+     *
+     *     // Start a new (child) Span:
+     *     var child = Tracer.startSpan('Subroutine', {
+     *         reference: Tracer.childOf(parent.context()),
```

Comment:
@bcronin I can see many (good) arguments for building convenience APIs above this layer... I was also thinking about allowing the caller to pass in a `Span` or a `SpanContext` for references (and using RTTI to convert to a `SpanContext` as needed).

> I'd argue rather strongly the `parent : <parentObject>` call should still be allowed if that is not a violation of the OpenTracing spec.

If we were to do that in the spirit of the current OT nomenclature, perhaps we'd get rid of the `reference: r` shorthand for `references: [r]` and instead have `childOf: spanOrSpanContext`. I.e.,

```
Tracer.startSpan('subroutine', { childOf: parent });
```

Thoughts?